kie-dmn-validation/src/main/java/org/kie/dmn/validation/DMNValidatorImpl.java

```diff
@@ -0,0 +1,177 @@
+    static {
+        try {
+            schema = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI).newSchema();
+        } catch (SAXException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    /**
+     * A KieContainer is normally available,
+     * unless at runtime some problem prevented building it correctly.
+     */
+    private Optional<KieContainer> kieContainer;
+    /**
+     * Collect at init time the runtime issues which prevented to build the `kieContainer` correctly.
+     */
+    private List<ValidationMsg> failedInitMsg;
+    
+    public DMNValidatorImpl() {
+        KieServices ks = KieServices.Factory.get();
+        KieFileSystem kfs = ks.newKieFileSystem();
+        ReleaseId rid = ks.newReleaseId("org.kie", "kie-dmn-validation", "1");
+        kfs.generateAndWritePomXML(rid);
+        try {
+            kfs.write("src/main/resources/rules.drl", IoUtils.readBytesFromInputStream( DMNValidatorImpl.class.getResourceAsStream("/rules.drl") ));
+        } catch (IOException e) {
+            LOG.error("Unable to read embedded DMN validation rule file.", e);
+            failedInitMsg.add(new ValidationMsg(Severity.ERROR, Msg.FAILED_VALIDATOR, e.getMessage()));
+        }
+        KieBuilder kieBuilder = ks.newKieBuilder(kfs).buildAll();
+        Results results = kieBuilder.getResults();
+        for ( Message m : results.getMessages(new Message.Level[]{Message.Level.ERROR, Message.Level.WARNING}) ) {
+            if (m.getLevel() == Level.ERROR) {
+                LOG.error("{}", m);
+            } else if (m.getLevel() == Level.WARNING) {
+                LOG.warn("{}", m);
+            }
+        }
+        
+        if (results.hasMessages(new Message.Level[]{Message.Level.ERROR})) {
+            LOG.error("Errors while compiling embedded DMN validation rules.");
+            results.getMessages().stream()
+                .map(m -> new ValidationMsg( Severity.ERROR, Msg.FAILED_VALIDATOR, m ))
+                .forEach(vm -> failedInitMsg.add(vm));
+            this.kieContainer = Optional.empty();
+        } else {
+            this.kieContainer = Optional.of( ks.newKieContainer(rid) );
+        }
+    }
+    
+    public void dispose() {
+        kieContainer.ifPresent( KieContainer::dispose );
+    }
+
+    @Override
+    public List<ValidationMsg> validateSchema(File xmlFile) {
+        List<ValidationMsg> problems = new ArrayList<>();
+        Source s = new StreamSource(xmlFile);
+        try {
+            schema.newValidator().validate(s);
+        } catch (SAXException | IOException e) {
+            problems.add(new ValidationMsg(ValidationMsg.Severity.ERROR, Msg.FAILED_XML_VALIDATION, e));
+        }
+        // TODO detect if the XSD is not provided through schemaLocation, and validate against embedded
+        return problems;
+    }
+    
+    @Override
+    public List<ValidationMsg> validateModel(Definitions dmnModel) {
+        if (!kieContainer.isPresent()) {
```

Comment:
if for some reason the "embedded" KieContainer could have not been created, the issue will be returned safely.